core/navigator.ts

```diff
@@ -18,35 +22,41 @@ export class Navigator {
    * Map from classes to a corresponding ruleset to handle navigation from
    * instances of that class.
    */
-  private rules: RuleMap<any> = new Map();
+  protected rules: RuleList<any> = [
+    new BlockNavigationPolicy(),
+    new FieldNavigationPolicy(),
+    new ConnectionNavigationPolicy(),
+    new WorkspaceNavigationPolicy(),
+  ];
 
   /**
-   * Associates a navigation ruleset with its corresponding class.
+   * Adds a navigation ruleset to this Navigator.
    *
-   * @param key The class whose object instances should have their navigation
-   *     controlled by the associated policy.
    * @param policy A ruleset that determines where focus should move starting
-   *     from an instance of the given class.
+   *     from an instance of its managed class.
    */
-  set<T extends INavigable<T>>(
-    key: new (...args: any) => T,
-    policy: INavigationPolicy<T>,
-  ) {
-    this.rules.set(key, policy);
+  addNavigationPolicy(policy: INavigationPolicy<any>) {
+    this.rules.push(policy);
   }
 
   /**
    * Returns the navigation ruleset associated with the given object instance's
    * class.
    *
-   * @param key An object to retrieve a navigation ruleset for.
+   * @param current An object to retrieve a navigation ruleset for.
    * @returns The navigation ruleset of objects of the given object's class, or
    *     undefined if no ruleset has been registered for the object's class.
    */
-  private get<T extends INavigable<T>>(
-    key: T,
-  ): INavigationPolicy<T> | undefined {
-    return this.rules.get(key.getClass());
+  private get(
+    current: IFocusableNode,
+  ): INavigationPolicy<typeof current> | undefined {
+    for (const rule of this.rules) {
```

Comment:
Optional, but could also be replaced with a find, e.g.:

```typescript
return this.rules.find((rule) => rule.isApplicable(current));
```

Reply:
Done